File: dotnet/DesignPatterns/BehavioralDesignPatterns/ChainOfResponsibilityDesignPattern/ChainOfResponsibilityDesignPattern.Example3/Handlers/TeamLeadHandler.cs
namespace ChainOfResponsibilityDesignPattern.Example3.Handlers;

public class TeamLeadHandler : BaseSupportHandler
{
    public TeamLeadHandler() : base("Team Lead") { }

    public override void HandleTicket(SupportTicket ticket)
    {
        // Team Lead can handle High priority tickets
        if (CanHandle(ticket, TicketPriority.High))
        {
            ticket.Resolve(HandlerName, "Coordinated with technical team and implemented custom solution.");
        }
        else
        {
            base.HandleTicket(ticket);
        }
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/ChainOfResponsibilityDesignPattern/ChainOfResponsibilityDesignPattern.Example3/src/Handlers/TeamLeadHandler.ts
import { SupportTicket, TicketPriority } from "../SupportTicket";
import { BaseSupportHandler } from "../BaseSupportHandler";

export class TeamLeadHandler extends BaseSupportHandler {
  constructor() { super("Team Lead"); }

  handleTicket(ticket: SupportTicket): void {
    if (this.canHandle(ticket, TicketPriority.High)) {
      ticket.resolve(this.handlerName, "Coordinated with technical team and implemented custom solution.");
    } else {
      super.handleTicket(ticket);
    }
  }
}
